### PR TITLE
Fix security vulnerabilities in JSON handling and decompression

### PR DESCRIPTION
**Pull request by rurban**
## Summary by cubic
Hardened decompression, JSON output, and build helpers to close security gaps, with new regression tests to lock in fixes. Prevents OOB reads, JSON injection, temp-file races, unverified downloads, and UB in pointer checks.

- **Bug Fixes**
  - R2007 decompression: pass src_end to read_instructions/read_literal_length, bounds-check every read, return int and propagate errors, fix CHECK_SRC to use src_end - *src < n, and expose decompress_r2007.
  - JSON: escape U+0000–U+001F and use json_cquote for 3DSOLID ACIS text to avoid unescaped quotes/backslashes.
  - dwgfilter: use mktemp; add EXIT/INT/TERM traps; in --debug, keep and print the JSON path.
  - Tests: add two OOB-read regressions for decompress_r2007; fix R2004 test buffer sizes; make main return non-zero on failures.

- **Dependencies**
  - Verify SHA-256 for libxslt and pslib downloads.
  - Vendor svg11-relaxng-20060604.zip; prefer local copy, else HTTPS with the same hash check; unzip and remove.

<sup>Written for commit 67422297c2d7dcf1635c62f484a5ce0e2ca368b7. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/LibreDWG/libredwg/pull/1260?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: build-aux/install-relaxng-svg11 [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+ZIP=svg11-relaxng-20060604.zip
+SHA256=d744594b116b6722b15ff65f57075f1f6692166eb2d2ab9111c00512bcf5d83f
+SRCDIR="$(dirname "$0")"
+
+# Prefer the vendored copy; fall back to download over HTTPS with hash check.
+if [ -f "$SRCDIR/$ZIP" ]; then
+    cp "$SRCDIR/$ZIP" "$ZIP"
+else
+    wget -q "https://yupotan.sppd.ne.jp/relax-ng/svg11/$ZIP"
+fi
```
[the remark this answers is not in the document]
Comment (rurban) [reply build-aux/install-relaxng-svg11:13]: No, we never execute from build-aux.

File: src/out_json.c [hunk 2]
```diff
@@ -1647,12 +1647,16 @@ json_3dsolid (Bit_Chain *restrict dat, const Dwg_Object *restrict obj,
...
               if ((*p == '\r' || *p == '\n') && p - s < 256)
                 {
-                  FIRSTPREFIX fprintf (dat->fh, "\"%.*s\"", (int)(p - s), s);
-                  // json_cquote (buf, s, p - s, dat->codepage));
+                  const char saved = *p;
+                  *(char *)p = '\0';
+                  FIRSTPREFIX fprintf (
+                      dat->fh, "\"%s\"",
+                      json_cquote (buf, s, sizeof (buf), dat->codepage));
+                  *(char *)p = saved;
```
[the remark this answers is not in the document]
Comment (rurban) [reply src/out_json.c:1659]: no, that's only in out_json.c. Which always acts on a copy.
